mpd/base.py

```diff
@@ -232,6 +237,10 @@ def _parse_objects(self, lines, delimiters=[]):
         if obj:
             yield obj
 
+    # Use this instead of _parse_objects whenever the result is returned
+    # immediately in a command implementation
+    _parse_objects_direct = _parse_objects
```

Comment:
A command is direct iff it is of the form `def X(self, ...): return self._Y_direct(...)`, and I'd add a `_Y_direct` variant to all `_Y` parser functions so that all commands that need to deal with arbitrarily many lines are direct. For those, it can be useful to have first results as they arrive (before the command has completed, eg. to populate a user interface with titles).